x-pack/solutions/observability/plugins/apm/public/components/shared/table_search_bar/table_search_bar.tsx

```diff
@@ -22,6 +23,11 @@ export function TableSearchBar({
   onChangeSearchQuery,
   techPreview = false,
 }: Props) {
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const debouncedSearchQuery = useCallback(debounce(onChangeSearchQuery, 500), [
```

Comment:
We have `useDebounce` from 'react-use/lib/useDebounce' so here we can use it and do something like:
```
  const [searchValue, setSearchValue] = useState(searchQuery ?? '');
...
  useDebounce(() => onChangeSearchQuery(searchValue), 500, [searchValue]
... 

Use those in `EuiFieldSearch`
```
Probably there are better examples in the code, I think in lens code I saw some. 

Wdyt?